.eslintrc.js

```diff
@@ -152,8 +152,21 @@ module.exports = {
 			},
 			{
 				selector: 'CallExpression[callee.object.name="page"][callee.property.name="waitFor"]',
-				message: 'Prefer page.waitForSelector instead.'
-			}
+				message: 'Prefer page.waitForSelector instead.',
+			},
+			{
+				// The <DotTip> component uses the `id` prop for something that does not require an
+				// instanceId; maybe we should change its key.
+				selector: 'JSXOpeningElement[name.name!="DotTip"] JSXAttribute[name.name="id"][value.type="Literal"]',
```

Comment:
Not sure how I feel about the exception here. Conversely, it would not be good to push this on to the responsibility of the developer rendering DotTip. I think I'd be okay with this as-is, but if we get to a point where we want these to become more of a generalized style standard, we might want to reevaluate how we approach this.

Reply:
Another idea, though imperfect itself, could be to detect whether the prop on which the `id` is assigned is a built-in DOM element (lowercase) or custom component. Not sure this is something we can do with the selector system (vs. a more formal ESLint plugin), and it won't catch cases where in a custom component we pass through props to a DOM element (`<Button id="" />`), so okay to punt for later consideration.

Reply:
I definitely wasn't a fan of this when I wrote it, and actually looking at it again I think it'd be better to `// eslint-disable-next-line` for `DotTip`'s `id` prop for now, and look at changing `DotTip`'s `id` prop to something else.

Reply:
Actually, yeah, it's worse to require any usage of `DotTip` to ignore an eslint rule. Even though it's only four instances, it's silly; I've filed #10305